client/src/App.js

```diff
@@ -1,241 +1,32 @@
-      headers: {
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-      },
-      method: 'POST',
-      body: JSON.stringify(log)
-    });
-  }
-
-  onWorkshopCodeDone(nextPhase, workshopCode) {
-    this.setState({workshopCode, phase: nextPhase});
-  }
-
-  onTitleDone(nextPhase) {
-    this.setState({phase: nextPhase});
-  }
-
-  onDataLoaded(loadedData) {
-    const {cohortNumber, students} = loadedData;
-    this.setState({cohortNumber, students});
-    this.doPrefetchStudentImages(students);
-  }
-
-  onDataError(err) {
-    error(err);
-  }
-
-  // Log the interaction, along with context about the session.
-  onInteraction(interaction) {
-    const {logs} = this.state;
-    const session = this.session();
-    const log = Log.create(session, interaction);
-    this.setState({ logs: logs.concat(log) });
-    this.doLog(log);
-  }
-
   render() {
-    const {isCodeOrg} = this.state;
-    return (
-      <div className="App">
-        <MobileSimulator minWidth={800} minHeight={400}>
-          {isCodeOrg ? this.renderCodeOrg() : this.renderDemo()}
-        </MobileSimulator>
-      </div>
-    );
-  }
-
-  // From code.org Code Studio, with identifier on query string
-  renderCodeOrg() {
-    const {phase, students} = this.state;
-    if (phase === Phases.WORKSHOP_CODE) return this.renderWorkshopCode(Phases.TITLE);
-    if (phase === Phases.TITLE) return this.renderTitle(Phases.CONSENT);
-    if (phase === Phases.CONSENT) return this.renderConsent(Phases.INTRODUCTION);
-    if (phase === Phases.INTRODUCTION) return this.renderIntroduction(Phases.STUDENTS);
-    if (!students) return this.renderLoading();
-    if (phase === Phases.STUDENTS) return this.renderStudents(Phases.DISCUSS);
-    if (phase === Phases.DISCUSS) return this.renderDiscuss(Phases.REVIEW);
-    if (phase === Phases.REVIEW) return this.renderReview(Phases.THANKS);
-    if (phase === Phases.THANKS) return this.renderThanks();
-  }
-
-  // Publicly open demo
-  renderDemo() {
-    const {phase, students} = this.state;
-    if (phase === Phases.TITLE) return this.renderTitle(Phases.INTRODUCTION);
-    if (phase === Phases.INTRODUCTION) return this.renderIntroduction(Phases.STUDENTS);
-    if (!students) return this.renderLoading();
-    if (phase === Phases.STUDENTS) return this.renderStudents(Phases.REVIEW);
-    if (phase === Phases.REVIEW) return this.renderReview(Phases.THANKS);
-    if (phase === Phases.THANKS) return this.renderThanks();
-  }
-
-  renderWorkshopCode(phase) {
-    return <WorkshopCode
-      shouldWarnAboutCodeStudio={this.shouldWarnAboutCodeStudio()}
-      onInteraction={this.onInteraction}
-      onDone={this.onWorkshopCodeDone.bind(this, phase)} />;
-  }
-
-  renderTitle(phase) {
     return (
-      <Lifecycle componentWillMount={() => this.doFetchData()}>
-        <Title
-          onInteraction={this.onInteraction}
-          onDone={this.onTitleDone.bind(this, phase)} />
-      </Lifecycle>
+      <BrowserRouter>
```

Comment:
This is the main change - and shows how to route a particular URL to a particular React component.